### ISSUE TITLE
Error when attempting to run this command: datacats pull

### ISSUE THREAD

**Issue by wide-awake** [datacats/datacats#312]
Full error:

atacats: problem was encountered:
Something that should not have happened happened when attempting to run this command:
     datacats pull

It is seems to be a bug.
Please report this issue to us by creating an issue ticket at

    https://github.com/datacats/datacats/issues

so that we would be able to look into that and fix the issue.

------------------------------
Please quote this traceback when reporting this error:
```
Traceback (most recent call last):
  File "/Users/stay-wide-awake/oasis/_env/lib/python2.7/site-packages/datacats/cli/main.py", line 93, in main
    return command_fn(opts)
  File "/Users/stay-wide-awake/oasis/_env/lib/python2.7/site-packages/datacats/cli/pull.py", line 46, in pull
    retrying_pull_image(i)
  File "/Users/stay-wide-awake/oasis/_env/lib/python2.7/site-packages/datacats/cli/pull.py", line 52, in retrying_pull_image
    "Failed to pull image {}.".format(image_name))
  File "/Users/stay-wide-awake/oasis/_env/lib/python2.7/site-packages/datacats/cli/pull.py", line 72, in _retry_func
    func(param)
  File "/Users/stay-wide-awake/oasis/_env/lib/python2.7/site-packages/datacats/cli/pull.py", line 83, in pull_image
    for s in pull_stream(image_name):
  File "/Users/stay-wide-awake/oasis/_env/lib/python2.7/site-packages/datacats/docker.py", line 380, in pull_stream
    return (json.loads(s) for s in _get_docker().pull(image, stream=True))
  File "/Users/stay-wide-awake/oasis/_env/lib/python2.7/site-packages/datacats/docker.py", line 92, in _get_docker
    api_version = version_client.version()['ApiVersion']
  File "/Users/stay-wide-awake/oasis/_env/lib/python2.7/site-packages/docker/client.py", line 837, in version
    return self._result(self._get(url), json=True)
  File "/Users/stay-wide-awake/oasis/_env/lib/python2.7/site-packages/docker/clientbase.py", line 86, in _get
    return self.get(url, **self._set_request_timeout(kwargs))
  File "/Users/stay-wide-awake/oasis/_env/lib/python2.7/site-packages/requests/sessions.py", line 477, in get
    return self.request('GET', url, **kwargs)
  File "/Users/stay-wide-awake/oasis/_env/lib/python2.7/site-packages/requests/sessions.py", line 465, in request
    resp = self.send(prep, **send_kwargs)
  File "/Users/stay-wide-awake/oasis/_env/lib/python2.7/site-packages/requests/sessions.py", line 573, in send
    r = adapter.send(request, **kwargs)
  File "/Users/stay-wide-awake/oasis/_env/lib/python2.7/site-packages/requests/adapters.py", line 415, in send
    raise ConnectionError(err, request=request)
ConnectionError: ('Connection aborted.', error(2, 'No such file or directory'))
```

**Comment by JackMc** [comment]
It looks like you're running on a Mac.

Please try running ``eval "$(boot2docker shellinit)"`` after initializing your boot2docker VM and starting it (``boot2docker init && boot2docker up``).

**Comment by wide-awake** [comment]
Yup, OSX 10.10. Will try and give feedback .

**Comment by JackMc** [comment]
We will work on a better error message to make this better

**Comment by wide-awake** [comment]
Still don't figure out what I had to do.
``` eval "$(boot2docker shellinit)"```  returns  ```command not found: boot2docker```

**Comment by wide-awake** [comment]
And also, when i try to install boot2docker on Docker: 

```Unable to find image 'boot2docker:latest' locally
Pulling repository docker.io/library/boot2docker
Error: image library/boot2docker:latest not found```

**Comment by JackMc** [comment]
You can install boot2docker using [this package](https://github.com/boot2docker/osx-installer/releases/download/v1.8.0/Boot2Docker-1.8.0.pkg)

**Comment by deniszgonjanin** [comment]
@wide-awake interesting - you seem to have the docker client installed, but you are not using `boot2docker`?

Can you confirm whether docker works or not, by running this command:
```
docker run busybox ls
```

**Comment by wide-awake** [comment]
@deniszgonjanin it's strange. I have installed docker using standard documentation for OS X. Your command give's me 
```
➜   docker run busybox ls
Post http:///var/run/docker.sock/v1.20/containers/create: dial unix /var/run/docker.sock: no such file or directory.
* Are you trying to connect to a TLS-enabled daemon without TLS?
* Is your docker daemon up and running?
```

Even thought, I have docker running in second terminal windows:

```

                        ##         .
                  ## ## ##        ==
               ## ## ## ## ##    ===
           /"""""""""""""""""\___/ ===
      ~~~ {~~ ~~~~ ~~~ ~~~~ ~~~ ~ /  ===- ~~~
           \______ o           __/
             \    \         __/
              \____\_______/


docker is configured to use the default machine with IP 192.168.99.100
For help getting started, check out the docs at https://docs.docker.com

bash-3.2$ 
```

**Comment by deniszgonjanin** [comment]
I see. Right now Docker on OSX is transitioning from [boot2docker](http://boot2docker.io/) as the official supported way to run on OSX, to the [docker toolbox](https://docs.docker.com/installation/mac/).

datacats assumes in certain places that if you are running on OSX that you are using `boot2docker`. We are working on moving to supporting docker toolbox as well (see #300) for the next release of datacats. For now, I would recommend one of two options to get datacats running:
a. Install boot2docker instead of the docker toolbox
b. Use a Linux VM

**Comment by wardi** [comment]
fixed by #317
